src/org/labkey/test/tests/list/ListTest.java

```diff
@@ -1812,6 +1812,210 @@ private void testTricky(String listName, boolean autoKey) throws IOException
 
     }
 
+    @Test // Issue 52098, Issue 49422
+    public void testNumberLikeLookup()
+    {
+        String lookupListName = TestDataGenerator.randomDomainName("lookupList");
+        String lookupKeyFieldName = TestDataGenerator.randomFieldName("lookupKeyField");
+        String lookupFieldName = TestDataGenerator.randomFieldName("lookupField");
+        log("Create a list to use as a lookup table with some number-like names.");
+        _listHelper.createList(PROJECT_VERIFY, lookupListName, lookupKeyFieldName,
+                new FieldDefinition(lookupFieldName, ColumnType.String));
+        String bulkData = lookupFieldName + "\n" +
+                "1E2\n" +
+                "102\n" +
+                "Lookup\n" +
+                ".123";
+        _listHelper.bulkImportData(bulkData);
+        DataRegionTable dataRegionTable = new DataRegionTable("query", getDriver());
+        CustomizeView customizer = dataRegionTable.openCustomizeGrid();
+        customizer.showHiddenItems();
+        customizer.addColumn(EscapeUtil.fieldKeyEncodePart(lookupKeyFieldName));
+        customizer.clickViewGrid();
+        List<Map<String, String>> actualValues = dataRegionTable.getTableData();
+        String keyNumber = actualValues.get(0).get(EscapeUtil.fieldKeyEncodePart(lookupKeyFieldName));
+        _listHelper.insertNewRow(Map.of(lookupFieldName, keyNumber));
+
+        log("Create a second list that looks up to the first list without the `Ensure value` checkbox selected.");
+        String baseListName = TestDataGenerator.randomDomainName("base");
+        String baseKeyFieldName = TestDataGenerator.randomFieldName("base Key Field");
+        String baseLookupFieldName = TestDataGenerator.randomFieldName("base Lookup Field");
+        String baseLookupFieldKey = EscapeUtil.fieldKeyEncodePart(baseLookupFieldName);
+        _listHelper.createList(PROJECT_VERIFY, baseListName, baseKeyFieldName);
+        EditListDefinitionPage listDefinitionPage = _listHelper.goToEditDesign(baseListName);
+        listDefinitionPage.getFieldsPanel()
+                .addField(baseLookupFieldName)
+                .setType(ColumnType.Lookup)
+                .setLookup(new FieldDefinition.IntLookup("lists", lookupListName));
```

Comment:
Regarding your question about `ColumnType.Lookup` being deprecated. `setType(Lookup)` isn't actually necessary; `setLookup` will set the type correctly.
```suggestion
                .addField(baseLookupFieldName)
                .setLookup(new FieldDefinition.IntLookup("lists", lookupListName));
```
Alternatively, you can pass a `FieldDefinition` to `addField`.

Reply:
Thanks. Done.